pyTigerGraph/gds/featurizer.py

```diff
@@ -463,34 +467,83 @@ def runAlgorithm(
+                    elif schema_type == "EDGE" and "e_type" in params:
+                        if isinstance(params["e_type"], str):
+                            schema_name = [params["e_type"]]
+                        elif isinstance(params["e_type"], list):
+                            schema_name = params["e_type"]
+                        else:
+                            raise ValueError("e_type should be either a list or string")
+
+                    global_types = []
+                    local_types = []
+                    if schema_type == "VERTEX":
+                        for v_type in schema_name:
+                            if "IsLocal" in self.conn.getVertexType(v_type, force=True):
+                                local_types.append(v_type)
+                            else:
+                                global_types.append(v_type)
+                    if schema_type == "EDGE":
+                        for e_type in schema_name:
+                            if "IsLocal" in self.conn.getEdgeType(e_type, force=True):
+                                local_types.append(e_type)
+                            else:
+                                global_types.append(e_type)
+
+                    if len(global_types) > 0 or global_schema:
                         _ = self._add_attribute(
                             schema_type,
                             feat_type,
                             feat_name,
-                            schema_name,
-                            global_change=global_schema,
+                            global_types,
+                            global_change=True,
                         )
-                else:
-                    query_ulr = self.algo_paths[query_name][-1]
-                    raise ValueError(
-                        "The algorithm does not provide any feature, see the algorithm details:"
-                        + query_ulr
-                        + "."
-                    )
-            result = self.conn.runInstalledQuery(
-                query_name, params, timeout=timeout, sizeLimit=sizeLimit, usePost=True
-            )
-            if result != None:
-                return result
+                    if len(local_types) > 0 or not(global_schema):
+                        _ = self._add_attribute(
+                            schema_type,
+                            feat_type,
+                            feat_name,
+                            local_types,
+                            global_change=False,
+                        )
+            '''
+            else:
+                query_ulr = self.algo_paths[query_name][-1]
+                raise ValueError(
+                    "The algorithm does not provide any feature, see the algorithm details: "
+                    + query_ulr
+                    + "."
+                )
+            '''
+        if not(query_name in [x.split("/")[-1] for x in self.conn.getInstalledQueries().keys()]):
```

Comment:
what is it is a custom query_name

Reply:
This is checking if the query_name is already installed on the db

Reply:
sorry, there is a typo there. what IF it is a custom query_name? It won't be in DB and it will try to install it by name. And it will fail to find it in the repo and throw an error. Okay. I guess that is fine.

Reply:
Ah, I will add the logic of `and not(custom_query)`. Don't want to deal with installing that as well since it would require additional params.